MAKE/Makefile.carrington_gcc_openmpi

```diff
@@ -62,8 +62,10 @@ FLAGS =
 #GNU flags:
 CC_BRAND = gcc
 CC_BRAND_VERSION = 9.4.0
-CXXFLAGS += -O3 -fopenmp -funroll-loops -std=c++17 -W -Wall -Wno-unused -mavx -march=znver2 #-flto
+CXXFLAGS += -O3 -fopenmp -funroll-loops -std=c++17 -mavx -march=znver2 #-flto
 testpackage: CXXFLAGS = -O2 -fopenmp -funroll-loops -std=c++17 -mavx -march=znver2
+CXXFLAGS += -Wall -Wextra -Wno-unused
+CXXFLAGS += -D OMPI_SKIP_MPICXX
```

Comment:
From what I can gather, the C++ bindings are unused even for C++. Them being unmaintained is a reason for a ton of warnings.

Reply:
They were added to the MPI standard by some overzealous C++ advocate, who then promptly abandoned them. Since the C bindings work just fine, the whole rest of the MPI gremium decided to deprecate them very soon after.

Our code doesn't use C++ bindings anywhere anymore, either.

Reply:
Should this then go in Makefile itself instead of platform-specific files? Or in all platform-specific files?

Reply:
Yeah, moving to the makefile itself is a good idea. The lines don't need to be duplicated, `CXXFLAGS += ...` applies to all build targets as long as it's after the `testpackage: CXXFLAGS = ...` line

Reply:
At least where we use OpenMPI I suppose, but it won't do anything otherwise I assume.